Svc/FramingProtocol/FprimeProtocol.cpp

```diff
@@ -19,11 +19,11 @@ FprimeFraming::FprimeFraming(): FramingProtocol() {}
 
 FprimeDeframing::FprimeDeframing(): DeframingProtocol() {}
 
-void FprimeFraming::frame(const U8* const data, const U32 size, Fw::ComPacket::ComPacketType packet_type) {
-    FW_ASSERT(data != nullptr);
+void FprimeFraming::frame(const Fw::Buffer& data, const Fw::Buffer& context) {
+    FW_ASSERT(data.getData() != nullptr);
     FW_ASSERT(m_interface != nullptr);
     // Use of I32 size is explicit as ComPacketType will be specifically serialized as an I32
-    FpFrameHeader::TokenType real_data_size = size + ((packet_type != Fw::ComPacket::FW_PACKET_UNKNOWN) ? sizeof(I32) : 0);
+    FpFrameHeader::TokenType real_data_size = context.getSize() + data.getSize();
     FpFrameHeader::TokenType total = real_data_size + FpFrameHeader::SIZE + HASH_DIGEST_LENGTH;
     Fw::Buffer buffer = m_interface->allocate(total);
```

Comment:
There should be a check here to see if the buffer is empty since the buffer manager can run out of buffers. I don't think the check should be an assert since it wouldn't be appropriate to assert flight software on a lack of buffer manager buffers. Therefore, I think it should do an early return with an error code. This error code could be use by the framer and an evr could be outputted.

Reply:
Agreed wholeheartedly.  Although we should check for `.isValid()` in the latest F´  because when buffer manager has no buffers, an invalid buffer is returned.